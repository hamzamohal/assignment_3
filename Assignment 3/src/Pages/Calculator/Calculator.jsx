
import './Calculator.scss';
import React from 'react';
import padItems from '../../Data/padItems';

const Calculator = ({ screenText, calculate }) => (
  <div className="calculator">
    <h1 className="calculator-title">Lets do some math</h1>
    <div className="screen">{screenText}</div>
    <div className="pad">
      {
        padItems.map((item) => (
          <div
            role="presentation"
            key={item.sym}
            onClick={() => calculate(item.sym)}
            className={item.class}
          >
            {item.sym}
          </div>
        ))
      }
    </div>
  </div>
);

export default Calculator;
