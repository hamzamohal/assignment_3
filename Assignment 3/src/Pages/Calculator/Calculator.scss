.calculator {
  width: 700px;
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  text-align: center;
}

.calculator-title {
  margin-bottom: 1.5rem;
}

.screen {
  width: 100%;
  height: 5rem;
  padding: 0.75rem;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  background-color: #858694;
  color: black;
}

.pad {
  display: grid;
  grid-template: 1fr 1fr 1fr 1fr / 1fr 1fr 1fr 1fr;
  gap: 0.5rem;
  width: 100%;

  & > div {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #e0e0e0;
    color: black;
    height: 5rem;
    cursor: pointer;
  }

  & .operation {
    background-color: #f5913e;
  }

  & .zero {
    grid-column: 1 / 3;
    text-align: center;
  }
}
