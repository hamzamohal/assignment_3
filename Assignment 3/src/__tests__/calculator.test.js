import TestRenderer from 'react-test-renderer';
import Calculator from '../Pages/Calculator/Calculator';

describe('Test Calculator component', () => {
  it('should renders Calculator component', () => {
    const calculatorComponent = TestRenderer.create(<Calculator />).toJSON();
    expect(calculatorComponent).toMatchSnapshot();
  });
});
