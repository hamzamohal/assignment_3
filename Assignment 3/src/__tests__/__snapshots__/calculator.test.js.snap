// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator component should renders Calculator component 1`] = `
<div
  className="calculator"
>
  <h1
    className="calculator-title"
  >
    Lets do some math
  </h1>
  <div
    className="screen"
  />
  <div
    className="pad"
  >
    <div
      className="all_clear"
      onClick={[Function]}
      role="presentation"
    >
      AC
    </div>
    <div
      className="plus_neg"
      onClick={[Function]}
      role="presentation"
    >
      +/-
    </div>
    <div
      className="modular"
      onClick={[Function]}
      role="presentation"
    >
      %
    </div>
    <div
      className="operation divide"
      onClick={[Function]}
      role="presentation"
    >
      /
    </div>
    <div
      className="seven"
      onClick={[Function]}
      role="presentation"
    >
      7
    </div>
    <div
      className="eight"
      onClick={[Function]}
      role="presentation"
    >
      8
    </div>
    <div
      className="nine"
      onClick={[Function]}
      role="presentation"
    >
      9
    </div>
    <div
      className="operation multiply"
      onClick={[Function]}
      role="presentation"
    >
      x
    </div>
    <div
      className="four"
      onClick={[Function]}
      role="presentation"
    >
      4
    </div>
    <div
      className="five"
      onClick={[Function]}
      role="presentation"
    >
      5
    </div>
    <div
      className="six"
      onClick={[Function]}
      role="presentation"
    >
      6
    </div>
    <div
      className="operation substract"
      onClick={[Function]}
      role="presentation"
    >
      -
    </div>
    <div
      className="one"
      onClick={[Function]}
      role="presentation"
    >
      1
    </div>
    <div
      className="two"
      onClick={[Function]}
      role="presentation"
    >
      2
    </div>
    <div
      className="three"
      onClick={[Function]}
      role="presentation"
    >
      3
    </div>
    <div
      className="operation add"
      onClick={[Function]}
      role="presentation"
    >
      +
    </div>
    <div
      className="zero"
      onClick={[Function]}
      role="presentation"
    >
      0
    </div>
    <div
      className="point"
      onClick={[Function]}
      role="presentation"
    >
      .
    </div>
    <div
      className="operation equal"
      onClick={[Function]}
      role="presentation"
    >
      =
    </div>
  </div>
</div>
`;
